src/clinical/service-worker-thread/tasks.ts

```diff
@@ -250,25 +258,31 @@ function extractEntityDataFromDonors(
   let clinicalEntityData: EntityClinicalInfo[] = [];
 
   donors.forEach(d => {
-    Object.values(ClinicalEntitySchemaNames).forEach(entity => {
-      const isQueriedEntity = isEntityInQuery(entity, entityTypes);
-      const isRelatedEntity = getRequiredDonorFieldsForEntityTypes(entityTypes).includes(entity);
+    Object.values(ClinicalEntitySchemaNames).forEach(entityName => {
+      const isQueriedEntity = isEntityInQuery(entityName, entityTypes);
+      const isRelatedEntity = getRequiredDonorFieldsForEntityTypes(entityTypes).includes(
+        entityName,
+      );
       const requiresSampleRegistration =
-        entity === ClinicalEntitySchemaNames.REGISTRATION &&
+        entityName === ClinicalEntitySchemaNames.REGISTRATION &&
         (entityTypes.includes('donor') || entityTypes.includes('sampleRegistration'));
       const requiresSpecimens =
-        entity === ClinicalEntitySchemaNames.SPECIMEN && entityTypes.includes('donor');
+        entityName === ClinicalEntitySchemaNames.SPECIMEN && entityTypes.includes('donor');
 
       const isRequiredEntity =
         isQueriedEntity || isRelatedEntity || requiresSampleRegistration || requiresSpecimens;
 
-      const clinicalInfoRecords = isRequiredEntity ? getClinicalEntitySubmittedData(d, entity) : [];
+      const clinicalInfoRecords = isRequiredEntity
+        ? getClinicalEntitySubmittedData(d, entityName)
+        : [];
 
-      const relatedEntity = clinicalEntityData.find(entityData => entityData.entityName === entity);
+      const relatedEntity = clinicalEntityData.find(
+        entityData => entityData.entityName === entityName,
+      );
       if (relatedEntity) {
         relatedEntity.results = _.concat(relatedEntity.results, clinicalInfoRecords);
       } else {
-        const clinicalEntity = { entityName: entity, results: clinicalInfoRecords };
+        const clinicalEntity: EntityClinicalInfo = { entityName, results: clinicalInfoRecords };
```

Comment:
See, here I can totally support the assertion as valid and needed. It makes sense, unlike the case we discussed for React components.